internal/ingress/annotations/annotations.go

```diff
@@ -46,15 +46,15 @@ const (
 	DefaultIngressClass = "kong"
 )
 
-func validIngress(ingressAnnotationValue, ingressClass string) bool {
+func validIngress(ingressAnnotationValue, ingressClass string, skipEmptyAnnotation bool) bool {
 	// we have 2 valid combinations
 	// 1 - ingress with default class | blank annotation on ingress
 	// 2 - ingress with specific class | same annotation on ingress
 	//
 	// and 2 invalid combinations
 	// 3 - ingress with default class | fixed annotation on ingress
 	// 4 - ingress with specific class | different annotation on ingress
-	if ingressAnnotationValue == "" && ingressClass == DefaultIngressClass {
+	if ingressAnnotationValue == "" && !skipEmptyAnnotation && ingressClass == DefaultIngressClass {
```

Comment:
Can you please update the code comments according to this change?
This is a tricky logic and we should document it for future-selves.